shesha-reactjs/src/components/formDesigner/components/image/index.tsx

```diff
@@ -24,41 +30,53 @@ const ImageComponent: IToolboxComponent<IImageProps> = {
 
   Factory: ({ model }) => {
     const { data: formData } = useFormData();
+    const { formMode } = useForm();
+    const { backendUrl } = useSheshaApplication();
+
+    const hasDimensions = model?.height && model?.width;
+    
+    if (formMode === 'designer' && !hasDimensions) {
+      return (
+        <Alert
+          showIcon
+          message="ImageComponent not configured properly"
+          description="Please make sure that you've specified 'height and width' properties."
+          type="warning"
+        />
+      );
+    }
 
     return (
       <ConfigurableFormItem model={model}>
         {(value) => {
-          // TODO:: Update the settings such that an option in the settings is added so allow the form to pass url through value.
-          // TODO:: Add ImgWrapper component
-          const url: string = getString(model?.url, formData) || value;
+          const url: string = getString(model.url, formData);
+          const storedFileId: string = getString(model.storedFileId, formData) || (value?.id || value);
 
           return (
-            <div className="container">
-              <img
-                src={url}
-                alt="Avatar"
-                width={model?.width}
-                height={model?.height}
-                style={getStyle(model.style, formData)}
+            <StoredFileProvider baseUrl={backendUrl}>
+              <FileView
+                height={model.height}
+                width={model.width}
+                dataSource={model.dataSource}
+                styles={getStyle(model?.style, formData)}
+                url={url}
+                storedFileId={storedFileId}
               />
-            </div>
+            </StoredFileProvider>
           );
         }}
       </ConfigurableFormItem>
     );
   },
-  initModel: model => {
+  initModel: (model) => {
     const customModel: IImageProps = {
       ...model,
     };
     return customModel;
   },
-  migrator: (m) => m
-    .add<IImageProps>(0, (prev) => migratePropertyName(migrateCustomFunctions(prev) as IImageProps))
-    .add<IImageProps>(1, (prev) => migrateVisibility(prev))
```

Comment:
@nthangeniphumudzo please restore usage of migrateVisibility, looks like it was removed by the merge